### PR TITLE
feat: only use rollup output for build

### PR DESCRIPTION
**Pull request by chmelevskij**
Current build setup copies files into folder and then runs full yarn installation.

Since we are building everything based on the usage we don't need all the `node_modules` and friends.

This changes reduces final package size after unpack on mac by ~100mb. Need to verify other platforms

### Before

Screenshot 2023-10-08 at 20 26 58
Screenshot 2023-10-08 at 20 28 18

### After

Screenshot 2023-10-08 at 20 25 56
Screenshot 2023-10-08 at 20 27 29

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by ctzsnooze** [comment]
Problem (not just from this code) with Mac.
Download -> Betaflight-Configurator-macOS.zip (size 150.1Mb)
double-click this -> betaflight-configurator_10.10.0-debug-daf3709_macOS.dmg (size 150.6Mb)
double-click this -> disk image opens normally.
Is it possible to distribute the Mac version *not* as a zip of the dmg, but as the dmg itself?  ie, at the stage of zipping the archive, omit that step for the Mac version?
Normally, Mac applications are distributed as dmg's only.

**Comment by ctzsnooze** [comment]
When testing this PR, all the javascript files for Tabs, ie all those from `/js/tabs`, except for `receiver_msp.js` and `receiver_msp.map`, have been moved to the root directory.  Apart from the `receiver_msp` files, `/js/tabs/` is empty.

The tabs still work properly.

**Comment by nerdCopter** [comment]
Old versions of Configurator rtained the .js files inside the `./js/tabs/` folder. unsure when or why this changed.  Would prefer such.

**Comment by chmelevskij** [comment]
> Old versions of Configurator rtained the .js files inside the ./js/tabs/ folder. unsure when or why this changed. Would prefer such.

What is the value in that? We don't include the source for C code why should we do that for configurator?

I could adjust the rollup to keep that structure if we really do need it.

**Comment by chmelevskij** [comment]
It does, but it's bundled based on usage (`import/export`) not just on the fact that we have things in `node_modules`

I will look into slightly different builds for prod and debug. I don't think you can even open dev tools in production builds.

We can see the original structure of the files etc. because of the sourcemaps (those `.map` files in final output). Chrome then re-assembles back the file structure in devtools. It has nothing to do with the original files being there.

**Comment by haslinghuis** [comment]
@chmelevskij please have a look - as Flash Firmware is not available after loading firmware.

**Comment by chmelevskij** [comment]
Hey there, will try to find time for this today 🙈

**Comment by chmelevskij** [comment]
@haslinghuis, had to copy in worker script we use in flasher tab https://github.com/betaflight/betaflight-configurator/blob/master/src/js/workers/hex_parser.js seemed to work now. Please verify

**Comment by haslinghuis** [comment]
git clean -xdf - so have to rebuild :)

**Comment by haslinghuis** [comment]
image

image

**Review by haslinghuis — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]
* approving, briefly tested both vite and nwjs

**Pull request merged**
